authors/apps/articles/views.py

```diff
@@ -89,3 +86,84 @@ def delete(self, request, slug):
         return Response({
             'errors': 'sorry article with that slug doesnot exist'
         }, status=status.HTTP_404_NOT_FOUND)
+
+
+class LikeDislikeArticleAPIView(generics.GenericAPIView):
+    """
+    Expose endpoints allowing a user to like an article and undo a like
+    """
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def post(self, request, slug='', action='like'):
+        """
+        Add a user to the list of people that liked an article
+        :param action: the action- whether we are liking or disliking
+        :param request: The incoming request object
+        :param slug: The expected article slug
+        :return: success for the action, or the appropriate error on failure
+        """
+        article = model_helpers.get_single_article_using_slug(slug)
+
+        if article and action is 'like':
+            article.disliked_by.remove(request.user)
+            article.liked_by.add(request.user)
+            return Response({
+                'message': 'liked'
+            })
+
+        if article and action is 'dislike':
```

Comment:
Would an `elif` clause work in this case? 
Do you also make provision for a scenario when `article` and `action` is `None` or an invalid value apart from `like` or `dislike` especially in the tests

Reply:
- It would. Didn't think it would be as clear. Noted though.
- Agreed. This could improve. Especially in the tests.